jax/_src/scipy/stats/bernoulli.py

```diff
@@ -42,23 +42,30 @@ def cdf(k: ArrayLike, p: ArrayLike) -> Array:
   k, p = jnp._promote_args_inexact('bernoulli.cdf', k, p)
   zero, one, one_minus_p = _lax_const(k, 0), _lax_const(k, 1), _lax_const(k, 1-p)
   conds = [
-    jnp.isnan(k),
     lax.lt(k, zero),
     jnp.logical_and(lax.ge(k, zero), lax.lt(k, one)),
     lax.ge(k, one)
     ]
-  vals = [jnp.nan, zero, one_minus_p, one]
+  vals = [zero, one_minus_p, one]
   return jnp.where(
-    jnp.logical_and(lax.ge(p, zero), lax.le(p, one)),
-    jnp.select(conds, vals), jnp.nan
+    jnp.logical_or(
+      jnp.logical_or(jnp.isnan(k), jnp.isnan(p)),
+      jnp.logical_or(lax.lt(p, zero), lax.gt(p, one))),
+    jnp.nan, 
+    jnp.select(conds, vals)
```

Comment:
Is there a reason you nest `select` within `where`? Why not put all branches in the `jnp.select` call?